src/frequenz/sdk/model/_manager.py

```diff
@@ -0,0 +1,98 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Load, update, monitor and retrieve machine learning models."""
+
+import asyncio
+import logging
+import pickle
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Generic, TypeVar, cast
+
+from frequenz.channels.file_watcher import EventType, FileWatcher
+
+from frequenz.sdk._internal._asyncio import cancel_and_await
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+@dataclass
+class _Model(Generic[T]):
+    """Represent a machine learning model."""
+
+    data: T
+    path: Path
+
+
+class ModelManager(Generic[T]):
+    """Manage machine learning models with direct dictionary configuration."""
+
+    def __init__(self, model_config: dict[str, Path]) -> None:
```

Comment:
I would still call the mapping of model to path a *repository*.

```suggestion
    def __init__(self, repository: dict[str, Path]) -> None:
```

Or maybe `model_paths`, but `config` sounds weird in this context, for me it sounds like it would configure how often they are checked for changes or stuff like that, configuration for the manager itself.

Reply:
I think `model_paths` is better. I will change it to that.